processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -525,15 +362,15 @@ public int size()
       return size;
     }
 
-    protected T _get(ByteBuffer copyValueBuffer, int startOffset, int endOffset)
+    T bufferedIndexedGet(ByteBuffer copyValueBuffer, int startOffset, int endOffset)
     {
       final int size = endOffset - startOffset;
-      if (startOffset == endOffset) {
+      lastReadSize = size;
+      if (size == 0) {
         return null;
       }
+      copyValueBuffer.clear();
```

Comment:
Analysis showed that initially limit always equals to capacity, so it should be ok. Added comments in appropriate places.